File: src/main/java/io/github/dqualizer/dqlang/types/rqa/configuration/resilience/ResilienceTestArtifact.java
package io.github.dqualizer.dqlang.types.rqa.configuration.resilience;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.types.dam.Endpoint;
import io.github.dqualizer.dqlang.types.rqa.definition.Artifact;
import io.github.dqualizer.dqlang.types.rqa.definition.loadtest.ResponseMeasures;
import io.github.dqualizer.dqlang.types.rqa.definition.stimulus.Stimulus;
import lombok.*;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class ResilienceTestArtifact {

    private Artifact artifact;
    @NonNull
    private String description;
    private Stimulus stimulus;
    @JsonProperty("response_measure")
    private ResponseMeasures responseMeasure;
    private Endpoint endpoint;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/rqa/configuration/resilience/ResilienceTestArtifact.kt
package io.github.dqualizer.dqlang.types.rqa.configuration.resilience

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.dqualizer.dqlang.types.rqa.definition.resilience.ResilienceResponseMeasures
import io.github.dqualizer.dqlang.types.rqa.definition.resilience.stimulus.ResilienceStimulus

data class ResilienceTestArtifact(
  val name: String,

  val description: String? = null,

  @JsonProperty("process_artifact")
  val processArtifact: ProcessArtifact? = null,

  @JsonProperty("cmsb_artifact")
  val cmsbArtifact: CmsbArtifact? = null,

  val stimulus: ResilienceStimulus? = null,

  @JsonProperty("response_measure")
  val responseMeasure: ResilienceResponseMeasures? = null
)
